src/core/app.lua

```diff
@@ -119,66 +122,134 @@ end
 -- new app network by making the changes needed.
 function configure (new_config)
    local actions = compute_config_actions(configuration, new_config)
-   apply_config_actions(actions, new_config)
-   configuration = new_config
```

Comment:
A very good question!

I guess I would first note that what we are doing is a fundamentally place-oriented operation underneath -- we are mutating the graph of instantiated apps and the links between them, not creating an entirely new graph.  When we're done we can't use the old idea of what the apps and links are because it is no longer valid.

You are right though that don't have to mutate the description of the app graph that is passed in to `app.configure` or `app.compute_config_actions`.  Even though we are mucking with links and such, we could avoid mutating the configuration.  However consider: in a multiprocess configuration as in https://github.com/Igalia/snabb/blob/yang/src/apps/config/README.md, the follower would be left with an out-of-date `app.configuration`.  Maybe this is OK?  The follower is designed to not need to know about the current configuration; it just applies changes.  I kept in the mutation so that the follower would have an up-to-date `app.configuration` though.